gcovr/configuration.py

```diff
@@ -1251,9 +1251,16 @@ def merge_options_and_set_defaults(
         "exclude_function_lines",
         ["--exclude-function-lines"],
         group="gcov_options",
-        help="Exclude coverage from lines defining a function Default: {default!s}.",
+        help="Exclude coverage from lines defining a function. Default: {default!s}.",
         action="store_true",
     ),
+    GcovrConfigOption(
+        "exclude_noncode_lines",
+        ["--no-exclude-noncode-lines"],
+        group="gcov_options",
+        help="Exclude coverage from lines which seem to be non-code. Default: {default!s}.",
+        action="store_false",
+    ),
```

Comment:
This comes a bit late, but note that options that are phrased as negations should be avoided.

1. Negations can be more complicated to understand for users, especially here were we have a triple negation “no”, “exclude”, “noncode”.
2. This interacts badly with config files.

For example, a config file entry `no-exclude-noncode-lines = no` is incomprehensible to me. The more I think about it, the less sure I am what is happening. As a user, I would also be confused by the help message:

```none
--no-exclude-noncode-lines
    Exclude coverage from lines which seem to be non-code.
    Default: True.
```

Does this mean that the exclusion is enabled (true), or that the option (no-exclude) is enabled by default?

Here, this could be avoided by phrasing the option positively, and auto-generating a negation:

```python
    GcovrConfigOption(
        "exclude_noncode_lines",
        ["--exclude-noncode-lines"],
        group="gcov_options",
        help="Exclude coverage from lines that don't seem to contain any code. Default: {default!s}.",
        action="store_true",
        const_negate=False, # <-- autogenerate --no-FLAG option
    ),
```

Alternatively, if a `--exclude-noncode-lines` option should never be allowed, it would be sufficient here to explicitly name the config key via `config="exclude-noncode-lines"`.

Reply:
I had also my problems with the name but the default is to remove lines which seems to have no code.
The easiest way would be to change our default to not remove them which can result in uncovered lines for the users. But then the user will get the line coverage which is also shown by gcov and this seems to be better.

Reply:
@latk I've tried your suggestion but it's only possible if the noncode lines are not excluded by default anymore. Our Command line parser doesn't allow a default for `action="sore_true"` (`AssertionError: action=store_true and default conflict`):
```python
    GcovrConfigOption(
        "exclude_noncode_lines",
        ["--exclude-noncode-lines"],
        config="exclude-noncode-lines",
        group="gcov_options",
        help="Exclude coverage from lines which seem to be non-code. Default: {default!s}.",
        action="store_true",
        const_negate=False,
        default=True,
    ),
```

I can go with this because than the default coverage numbers are closed to the numbers of gcov. Shall I do this breaking change?